doc/devdocs/types.rst

```diff
@@ -0,0 +1,522 @@
+****************
+More about types
+****************
+
+.. currentmodule:: Base
+
+If you've used Julia for a while, you understand the fundamental role
+that types play.  Here we try to get under the hood, focusing
+particularly on :ref:`parametric types <man-parametric-types>`.
+
+Types and sets (and ``Any`` and ``Union()``/``Bottom``)
+-------------------------------------------------------
+
+It's perhaps easiest to conceive of Julia's type system in terms of
+sets.  A concerete type corresponds to a single entity in the space of
+all possible types; an abstract type refers to a collection (set) of
+concrete types.  ``Any`` is a type that describes the entire universe
+of possible types; ``Integer`` is a subset of ``Any`` that includes
+``Int``, ``Int8``, and other concrete types.  Internally, julia also
+makes heavy use of another type known as ``Bottom``, or equivalently,
+``Union()``.  This corresponds to the empty set.
+
+Julia's types support the standard operations of set theory: you can
+ask whether ``T1`` is a "subset" (subtype) of ``T2`` with ``T1 <:
+T2``.  Likewise, you intersect two types using ``typeintersect`` and
+take their union with ``typejoin``::
+
+    julia> typeintersect(Int,Float64)
+    Union()
+
+    julia> typejoin(Int,Float64)
+    Real
+
+    julia> typeintersect(Signed, Union(UInt8, Int8))
+    Int8
+
+    julia> typejoin(Signed, Union(UInt8, Int8))
+    Integer
+
+    julia> typeintersect(Tuple{Integer,Float64}, Tuple{Int,Real})
+    Tuple{Int64,Float64}
+
+    julia> typejoin(Tuple{Integer,Float64}, Tuple{Int,Real})
+    Tuple{Integer,Real}
+
+While these operations may seem abstract, they lie at the heart of
+julia.  For example, method dispatch is implemented by stepping
```

Comment:
I think the style has been to prefer capitalizing Julia in the docs, except where referring to invoking the executable from the command line?